### PR TITLE
Run compiletest on rmc documentation code

### LINKED ISSUE #254: Training material
Infra:

* [X] Tools and CI "Hello World" for the documentation #456 
* [ ] Extracting and testing the examples from the documentation #635

Docs:

* [X] Getting started guide #471
* [X] Installation guide #471
* [X] Comparison with other tools initial documentation #471
* [X] Using RMC on a single file #525
* [X] Using cargo-rmc #554
* [X] Developer documentation #572
* Migration of CBMC tutorial docs to RMC
  * [X] #474
  * [X] #481
  * [X] #494
  * [X] #525

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

A few findings:

1. Compiletest does weird things and seem very insistent on the convention of putting everything under `src/test/` so I created a symlink. (Github doesn't seem to indicate this very obviously, but that's that `src/test/rmc-docs` is.)
2. Its behavior seems to just be `find . -name "*.rs"` and compile each file separately, but that's fine for the doc tests, since we're only using the `proptest` dependency if `cfg(test)` which we aren't here.
3. All but one doc test is `rmc-verify-fail` but that's fine I think since we mostly want to ensure they still build correctly. It'd be nice if we could be more precise about why they fail, but that's not easy to do, so oh well

### Resolved issues:

Resolves #254 

### Testing:

* How is this change tested? it's new tests

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: scripts/rmc-regression.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ check-cbmc-viewer-version.py --major 2 --minor 5
 ./scripts/setup/build_rmc_lib.sh
 
 # Standalone rmc tests, expected tests, and cargo tests
-./x.py test -i --stage 1 rmc firecracker prusti smack expected cargo-rmc
+./x.py test -i --stage 1 rmc firecracker prusti smack expected cargo-rmc rmc-docs
```
Comment (adpaco-aws) [inline scripts/rmc-regression.sh:28]: I thought testing `rmc-docs` examples was going to be done by the `build-docs.sh` script (see the `TODO` comment there).

I'm okay with having it here but then we should remove the comment. Does the regression workflow need to change too? Please check.

**Review by adpaco-aws — CHANGES_REQUESTED** [review]
LGTM modulo the comment below.

File: scripts/rmc-regression.sh (hunk 1 above)
Comment (tedinski) [reply scripts/rmc-regression.sh:28]: Ah good point! I'll remove the comment and add another bit of documentation about how to run the tests for the docs.

The CI workflow runs the `rmc-regression.sh` script.

Comment (adpaco-aws) [reply scripts/rmc-regression.sh:28]: Right, but I recall having a discussion with you about the condition to execute `build-docs.sh`. Anyway, looks like it executes on Ubuntu 20.04 only - that is fine if we run them from `rmc-regression.sh`.

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**